### DISCUSSION TITLE
Crash while loading custom textures

### DISCUSSION THREAD

**Discussion by Medard22** [Medard22/Dolphin-MMJR2-VBI#33] [Bug Report]
According to @ivan89el. I need more information to figure out what's wrong.

**Comment by Medard22** [comment]
So I tested loading custom textures in general and found no issues in both Vulkan and OpenGL. Tested on Mario Kart Double Dash (PAL) and  Mario Kart: Double Dash!! HD Texture Pack (GCN). The custom textures load perfectly fine (there is a striking improvement in texture quality).  I doubt it´s tied to this particular fork or build. I really don´t remember digging around custom texture code. There is either problem with the texture pack or some other issue. If @ivan89el won´t provide further information I will close this thread.

**Comment by Medard22** [comment]
I couldn´t reproduce the bug and since I was given no further information I am closing this thread.

**Comment by ivan89el** [comment]
Game NFS Underground, Textures: https://disk.yandex.ru/d/PESAtbScyjaI_w
It all works on all forks and official versions

**Reply by Medard22** [reply]
I can´t reproduce it on my devices. I downloaded the texture pack you provided and tested it on clean iso of NFS Underground (USA).  The custom textures loads fine and I experienced no crash. 
Truth is both my devices are running my latest devbuilds. I can send you an updated APK when I get back home. Just be warned that  savestates from previous stable versions and beta are NOT compatible.

**Reply by ivan89el** [reply]
Ok

**Comment by Medard22** [comment]
This is the latest dev build. Try it with that one. I ran a couple of laps, performance kind of sucks on my phone with custom textures, but no crashes so far.

https://drive.google.com/file/d/1ks8Zuw4_K0BjPhC-e1y_UnUAs7wrVAy_/view?usp=sharing

**Reply by ivan89el** [reply]
Unfortunately, the black screen is frozen. Also, other games do not start until I remove the "Load Custom Textures" or remove the texture pack.

**Reply by Medard22** [reply]
Could you tell me again what device are you using? Enabling logging and attaching dolphin.log would help too.

**Reply by ivan89el** [reply]
Anbernic RG405m Android 12 Unisoc T618 2 x Cortex-A75 2 GHz + 6 x Cortex-A55 1.8 GHz. GPU Mali G52MP2 4GB ram.
Log: https://disk.yandex.ru/d/C3VixGckhAfhIw

**Reply by Medard22** [reply]
I tested it on my RP3+ which has similar hardware and custom textures are loading fine. It must be something either specific to your device or you have something in your settings that is causing this problem. You could try to backup your mmjr2-vbi folder, delete it and start over with a clean state. Use default configuration, just copy over the custom textures and enable loading custom textures. I really cannot fix what I can´t reproduce and I really haven´t been poking anywhere near the custom texture code.

Anyway the performance of this game is rather bad. VBI Skip can make it playable, but with terrible framepacing issues. With additional performance cost of high resolution textures this game will never be enjoyable on a T618 device.

**Comment by ivan89el** [comment]
Can you move the custom MMJR2-VBI folder to Android/Data for the test?

**Reply by Medard22** [reply]
That would most likely require implementing scoped storage which MMJR2 is not using.
